src/compiler/checker.ts

```diff
@@ -4164,6 +4164,14 @@ namespace ts {
                         && isDeclarationVisible(declaration.parent)) {
                         return addVisibleAlias(declaration, declaration);
                     }
+                    else if (isBindingElement(declaration) && isInJSFile(declaration) && declaration.parent?.parent // exported import-like top-level JS require statement
+                        && isVariableDeclaration(declaration.parent.parent)
+                        && declaration.parent.parent.parent?.parent && isVariableStatement(declaration.parent.parent.parent.parent)
+                        && !hasSyntacticModifier(declaration.parent.parent.parent.parent, ModifierFlags.Export)

```

Comment:
should we care about checking for `const` or `require` here? *Probably* doesn't matter because I don't think there are any other declarations that look like `const { x } = require('y')`.

Reply:
Maybe. Might just be worth checking for `symbol.flags & SymbolFlags.Alias`.